FitGoal/FitGoal/Singup&Singin/SocialMediaAuthenticator.swift

```diff
@@ -53,4 +69,127 @@ class SocialMediaAuthenticator: NSObject, GIDSignInDelegate {
             completion?(.success(()))
         }
     }
+    
+    //MARK: - Facebook
+    
+    private func facebookSignIn(sender: UIViewController, completion: @escaping SignInCallback) {
+        let permissions = ["public_profile", "email"]
+        LoginManager().logIn(permissions: permissions, from: sender) { (logInResuts, error) in
+            if let error = error {
+                completion(.failure(error))
+            }
+            else {
+                guard let logInResuts = logInResuts else {
+                    let error = LoginError.noLogInResultsFound
+                    completion(.failure(error))
+                    return
+                }
+                
+                if logInResuts.isCancelled {
+                    let error = LoginError.userCanceledLogIn
+                    completion(.failure(error))
+                }
+                else {
+                    guard let user = Auth.auth().currentUser else {
+                        let error = MissingUserInfoError.noUserFound
+                        completion(.failure(error))
+                        return
+                    }
+                    
+                    guard let email = user.email else {
+                        let error = MissingUserInfoError.noEmailFound
+                        completion(.failure(error))
+                        return
+                    }
+                    
+                    Auth.auth().fetchSignInMethods(forEmail: email) { (signInMethods, error) in
+                        if let error = error {
+                            completion(.failure(error))
+                        }
+                        else if signInMethods != nil && !signInMethods!.contains("facebook.com") {
```

Comment:
Why are you force unwrapping the `signInMethods` variable? How can you be sure it won't be never `nil`.?